cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -0,0 +1,690 @@
+
+    // Kernels' grid sizes
+    BlockOffsetT init_grid_size         = DivideAndRoundUp(num_tiles, INIT_KERNEL_THREADS);
+    BlockOffsetT batch_memcpy_grid_size = num_tiles;
+
+    // Kernels
+    auto init_scan_states_kernel =
+      InitTileStateKernel<BLevBufferOffsetTileState, BLevBlockOffsetTileState, BlockOffsetT>;
+    auto bach_memcpy_non_blev_kernel = BatchMemcpyKernel<MaxPolicyT,
+                                                         InputBufferIt,
+                                                         OutputBufferIt,
+                                                         BufferSizeIteratorT,
+                                                         BufferOffsetT,
+                                                         BlevBufferSrcsOutItT,
+                                                         BlevBufferDstsOutItT,
+                                                         BlevBufferSizesOutItT,
+                                                         BlevBufferTileOffsetsOutItT,
+                                                         BlockOffsetT,
+                                                         BLevBufferOffsetTileState,
+                                                         BLevBlockOffsetTileState>;
+
+    auto multi_block_memcpy_kernel = MultiBlockBatchMemcpyKernel<MaxPolicyT,
+                                                                 BufferOffsetT,
+                                                                 BlevBufferSrcsOutItT,
+                                                                 BlevBufferDstsOutItT,
+                                                                 BlevBufferSizesOutItT,
+                                                                 BlevBufferTileOffsetsOutItT,
+                                                                 BLevBufferOffsetTileState,
+                                                                 BlockOffsetT>;
+
+    constexpr uint32_t BLEV_BLOCK_THREADS = ActivePolicyT::AgentLargeBufferPolicyT::BLOCK_THREADS;
+
+    // Get device ordinal
+    int device_ordinal;
+    if (CubDebug(error = cudaGetDevice(&device_ordinal)))
+      return error;
+
+    // Get SM count
+    int sm_count;
+    if (CubDebug(error = cudaDeviceGetAttribute(&sm_count,
+                                                cudaDevAttrMultiProcessorCount,
+                                                device_ordinal)))
+      return error;
+
+    // Get SM occupancy for the batch memcpy block-level buffers kernel
+    int batch_memcpy_blev_occupancy;
+    if (CubDebug(error = MaxSmOccupancy(batch_memcpy_blev_occupancy,
+                                        multi_block_memcpy_kernel,
+                                        BLEV_BLOCK_THREADS)))
+      return error;
+
+    uint32_t batch_memcpy_blev_grid_size =
+      static_cast<uint32_t>(sm_count * batch_memcpy_blev_occupancy);
```

Comment:
I'd also try:
```suggestion
      static_cast<uint32_t>(sm_count * batch_memcpy_blev_occupancy * CUB_SUBSCRIPTION_FACTOR(0));
```

Reply:
Will do. Can you fill me in on when we are supposed to use an oversubscription factor and why we would expect it to improve things? I couldn't find documentation on that.